### PR TITLE
StreamAnalytics schema and tests

### PR DESCRIPTION
**Pull request by thbagley**
- Generated Azure Stream Analytics Schema for API Version 2017-03-01. 

- Based on swagger spec https://github.com/Azure/azure-rest-api-specs/tree/current/specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/2016-03-01.

- Moved properties and added additionalProperties field throughout for correct validation

- Test cases for streaming jobs and component parts (input, output, etc)

### CODE REVIEW

**Comment by Tiano2017** [comment]
@thbagley  you need to add your new resource types to 2015-01-01/deploymentTemplate and 2014-04-01-preview/deploymentTemplate.

File: tests/2017-03-01/Microsoft.StreamAnalytics.Input.tests.json [hunk 1]
```diff
@@ -0,0 +1,800 @@
+{
+  "tests": [
+    {
+      "name": "Test Valid Stream Input",
+      "definition": "https://schema.management.azure.com/schemas/2016-03-01/Microsoft.StreamAnalytics.json#/definitions/Input",
```
Comment (TianoMS) [inline tests/2017-03-01/Microsoft.StreamAnalytics.Input.tests.json]: >2016-03-01 [](start = 65, length = 10)

this should be 2017-03-01.

**Comment by thbagley** [comment]
Thanks for pointing that out!

**Comment by thbagley** [comment]
Anything else I need to do? Thanks!

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 2]
```diff
@@ -375,6 +375,7 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2017-04-01/Microsoft.Relay.json#/resourceDefinitions/namespaces_wcfRelay_authorizationRules" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-04-01/Microsoft.Relay.json#/resourceDefinitions/namespaces_hybridConnection" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-04-01/Microsoft.Relay.json#/resourceDefinitions/namespaces_hybridConnection_authorizationRules" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-03-01/Microsoft.StreamAnalytics.json#/resourceDefinitions/streamingjobs" }
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >2017-03-01 [](start = 73, length = 10)

2016-03-01.

**Comment by thbagley** [comment]
That should resolve all of the API version errors... anything else?

**Pull request merged**